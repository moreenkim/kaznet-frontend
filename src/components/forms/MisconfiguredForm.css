.popover {
  max-width: 500px !important;
}
