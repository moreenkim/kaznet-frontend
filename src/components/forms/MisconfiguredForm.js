// Renders the detail page title section
import React, { Component } from 'react';
import FontAwesomeIcon from '@fortawesome/react-fontawesome';
import { Popover, PopoverHeader, PopoverBody } from 'reactstrap';

import MisconfiguredFormMessage from './MisconfiguredFormMessage';
import { MAIN_COLOR } from '../../constants';

import './MisconfiguredForm.css';

export default class MisconfiguredForm extends Component {
  constructor(props) {
    super(props);

    this.toggle = this.toggle.bind(this);
    this.state = {
      popoverOpen: false
    };
  }

  toggle() {
    this.setState({
      popoverOpen: !this.state.popoverOpen
    });
  }

  render() {
    return (
      <span>
        &nbsp;&nbsp;
        <FontAwesomeIcon
          id={`Popover-${this.props.form.id}`}
          icon="exclamation-circle"
          color={MAIN_COLOR}
          onClick={this.toggle}
          style={{ cursor: 'pointer' }}
        />
        <Popover
          placement="right"
          isOpen={this.state.popoverOpen}
          target={`Popover-${this.props.form.id}`}
          toggle={this.toggle}
        >
          <PopoverHeader>This form is not configured correctly</PopoverHeader>
          <PopoverBody>
            <MisconfiguredFormMessage />
          </PopoverBody>
        </Popover>
      </span>
    );
  }
}
