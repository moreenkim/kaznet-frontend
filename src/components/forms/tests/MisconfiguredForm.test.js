// test ExportModal
import React from "react";
import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";

import MisconfiguredForm from "../MisconfiguredForm";
import { formIdThreeById } from "../../../store/forms/tests/fixtures";

describe("components/forms/MisconfiguredForm", () => {
  it("renders without crashing", () => {
    shallow(<MisconfiguredForm form={formIdThreeById} />);
  });
  it("renders the MisconfiguredForm component", () => {
    const div = document.createElement("div");
    document.body.appendChild(div);
    const wrapper = mount(<MisconfiguredForm form={formIdThreeById} />, {
      attachTo: div
    });

    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper.unmount();
  });
});
