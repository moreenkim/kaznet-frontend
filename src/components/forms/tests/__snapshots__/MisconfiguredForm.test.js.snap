// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/forms/MisconfiguredForm renders the MisconfiguredForm component 1`] = `
<MisconfiguredForm
  form={
    Object {
      "attributes": Object {
        "created": "2018-06-21T12:15:23.175198+03:00",
        "deleted_at": null,
        "has_task": true,
        "id_string": "form_3",
        "last_updated": null,
        "metadata": Object {
          "configuration_status": "members_cannot_submit",
          "downloadable": true,
          "owner": "kaznet",
          "owner_url": "http://127.0.0.1:8000/api/v1/users/kaznet",
        },
        "modified": "2018-06-21T12:15:23.175232+03:00",
        "ona_pk": 6373,
        "project_id": 5522,
        "task_name": "Form 3 Task",
        "title": "Form 3",
      },
      "id": "3",
      "type": "XForm",
    }
  }
>
  <span>
      
    <FontAwesomeIcon
      border={false}
      className=""
      color="#711c27"
      fixedWidth={false}
      flip={null}
      icon="exclamation-circle"
      id="Popover-3"
      listItem={false}
      mask={null}
      name=""
      onClick={[Function]}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      style={
        Object {
          "cursor": "pointer",
        }
      }
      symbol={false}
      transform={null}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-exclamation-circle fa-w-16 "
        color="#711c27"
        data-icon="exclamation-circle"
        data-prefix="fas"
        id="Popover-3"
        onClick={[Function]}
        role="img"
        style={
          Object {
            "cursor": "pointer",
          }
        }
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zm-248 50c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </FontAwesomeIcon>
    <Popover
      delay={
        Object {
          "hide": 0,
          "show": 0,
        }
      }
      hideArrow={false}
      isOpen={false}
      placement="right"
      placementPrefix="bs-popover"
      target="Popover-3"
      toggle={[Function]}
    />
  </span>
</MisconfiguredForm>
`;
